Source/NSubstitute/Core/ISubstitutionContext.cs

```diff
@@ -9,6 +9,7 @@ public interface ISubstitutionContext
     {
         ISubstituteFactory SubstituteFactory { get; }
         SequenceNumberGenerator SequenceNumberGenerator { get; }
+        IPendingSpecificationInfo ThreadLocalPendingSpecificationInfo { get; set; }
```

Comment:
I think it's better to change naming from `ThreadLocalPendingSpecificationInfo` to `PendingSpecificationInfo`. It's an interface and doesn't know about implementation.